### PR TITLE
Refactors sound toggle prefs away from legacy toggles, introduces a new sound pref for jukeboxes

### PR DESCRIPTION
**Pull request by Thunder12345**
## About The Pull Request

Migrates sound toggle prefs away from legacy toggles, and changes all related preference checks to the modern form. Cleans up unnecessary defines linked to the old prefs. Increments the minimum save file version so a piece of old sound related code can be neatly removed. (The minimum version hasn't changed in two years anyone who hasn't played in at least that long probably doesn't even remember they had prefs saved)

Splits off jukebox music into its own preference distinct from instruments. (This was actually the change that brought about this whole PR the rest of the igration just made sense to do at the same time)

## Why It's Good For The Game

More granularity in sound preferences is good. It is quite reasonable for players to want to hear normal volume and largely unobtrusive instruments, without also being subjected to the aural torture of the jukebox. As implementing this wothout creating an unwelcome extra legacy toggle already required setting up a non-legacy sound toggle, it is most reasonable to migrate the other sounds at the same time.

## Changelog
:cl:
qol: Jukebox music can be controlled by a new seperate preference, and is no longer linked to the instrument sound preference option.
refactor: Sound related toggle preferences have been migrated away from the legacy system. Pre-existing preferences should be safely migrated but players are advised to check.
/:cl:

### CODE REVIEW

File: code/modules/client/preferences/middleware/legacy_toggles.dm [hunk 1]
```diff
@@ -110,13 +102,13 @@
...
 
-		// I know this looks silly, but this is the only one that cares
+		/*// I know this looks silly, but this is the only one that cares
 		// and NO NEW LEGACY TOGGLES should ever be added.
 		if (legacy_flag == SOUND_LOBBY)
 			if (value && isnewplayer(user))
 				user.client?.playtitlemusic()
 			else
-				user.stop_sound_channel(CHANNEL_LOBBYMUSIC)
+				user.stop_sound_channel(CHANNEL_LOBBYMUSIC)*/
```
Comment (Mothblocks) [inline code/modules/client/preferences/middleware/legacy_toggles.dm:111]: Don't comment code, delete it

File: code/modules/client/preferences/species_features/sounds.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/// Controls hearing dance machines
```
Comment (Mothblocks) [inline code/modules/client/preferences/species_features/sounds.dm:1]: This is inside the species_features folder, move it out to the other game preferences.

File: code/modules/client/preferences_savefile.dm [hunk 3]
```diff
@@ -42,9 +42,6 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 //if your savefile is 3 months out of date, then 'tough shit'.
 
 /datum/preferences/proc/update_preferences(current_version, savefile/S)
-	if(current_version < 33)
-		toggles |= SOUND_ENDOFROUND
```
Comment (Mothblocks) [inline code/modules/client/preferences_savefile.dm:46]: You don't need to increase the savefile version at all if this is what you are removing. `/datum/preference/toggle/sound_endofround` defaults to TRUE (I assume), which is the equivalent to this. Even though your savefile version isn't too destructive, it shouldn't be necessary and I think we should avoid it.

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/sounds.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import { CheckboxInput, FeatureToggle } from '../base';
+
+export const sound_dance_machine: FeatureToggle = {
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/sounds.tsx:3]: Put it in ABC order please.

File: code/modules/client/preferences/species_features/sounds.dm [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+/datum/preference/toggle/sound_dance_machine
```
Comment (Mothblocks) [inline code/modules/client/preferences/species_features/sounds.dm:2]: ABC order please

File: code/game/machinery/dance_machine.dm [hunk 6]
```diff
@@ -454,7 +454,7 @@
 		var/sound/song_played = sound(selection.song_path)
 
 		for(var/mob/M in range(10,src))
-			if(!M.client || !(M.client.prefs.toggles & SOUND_INSTRUMENTS))
+			if(!M.client || !(M.client.prefs.read_preference(/datum/preference/toggle/sound_dance_machine)))
```
Comment (Mothblocks) [inline code/game/machinery/dance_machine.dm:457]: If you change the preference, it should mute existing music, and vice versa.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Can you verify that a player's preferences for this are carried over after this change is made? If not you need to increase the savefile version and migrate them over.

File: code/modules/client/preferences/middleware/legacy_toggles.dm (hunk 1 above)
Comment (Thunder12345) [reply code/modules/client/preferences/middleware/legacy_toggles.dm:111]: I knew I forgot something...

**Comment by Thunder12345** [comment]
> Can you verify that a player's preferences for this are carried over after this change is made? If not you need to increase the savefile version and migrate them over.

How do you test that?

**Comment by Mothblocks** [comment]
It's trickier than it should be, but here's the best way to do it.

Load up master, and configure your settings with whatever toggles on or off.

Backup data/player_saves.

Check out your branch, recompile, and reload. See if your preferences are the same.

If this doesn't migrate automatically (I suspect it does not), you have to copy the sound toggle bitflags that exist now into the migration file (so nobody "refactors" it and reorders it), read the field directly, and write the new toggles.

**Comment by Thunder12345** [comment]
@Mothblocks full mothblockification has been achieved

**Comment by Mothblocks** [comment]
Will look over this soon but from a glance looks like a lot of my pressing concerns are fixed :+1:

**Pull request merged**